framework/build.sh

```diff
@@ -7,25 +7,27 @@ cd build
 
 cmake \
   -G Ninja \
-  -DUSE_SYSTEM_EIGEN=ON \
-  -DUSE_JEMALLOC=OFF \
-  -DENABLE_OPENGL=OFF \
-  -DENABLE_DOCS=OFF \
-  -DENABLE_MANTIDPLOT=FALSE \
-  -DENABLE_WORKBENCH=FALSE \
-  -DENABLE_OPENCASCADE=FALSE \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DUSE_JEMALLOC=ON \
+  -DUSE_CCACHE=OFF \
   -DPython_EXECUTABLE="$CONDA_PREFIX/bin/python" \
   -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
   -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX \
-  -DCMAKE_SKIP_INSTALL_RPATH=ON \
+  -DCMAKE_OSX_DEPLOYMENT_TARGET=$OSX_VERSION \
+  -DHDF5_ROOT=$CONDA_PREFIX \
+  -DCMAKE_OSX_SYSROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX$OSX_VERSION.sdk" \
+  -DUSE_SYSTEM_EIGEN=OFF\
+  -DENABLE_MANTIDPLOT=OFF\
+  -DENABLE_WORKBENCH=OFF\
+  -DENABLE_OPENGL=OFF\
+  -DENABLE_DOCS=OFF\
+  -DENABLE_OPENCASCADE=OFF\
+  -DUSE_PYTHON_DYNAMIC_LIB=OFF\
```

Comment:
I wonder if we should add a comment about this flag to say that conda Python builds on OSX statically link to the runtime so we disable dynamic linking to avoid stamping on the symbols?